### PR TITLE
Use bitwise operators instead of iterating over bits

### PR DESCRIPTION
**Pull request by b-viguier**
I detected a performance bottleneck when we check a big list of IPs. These 2 methods perform bitwise operations between IP, but **bit per bit**. It's really inneficient.

This solution use [Php bitwise operators](https://www.php.net/manual/en/language.operators.bitwise.php): IPs are split in several integers (the biggest possible integer according to `PHP_INT_MAX`) and operations are performed with native operators.

At most 2 iterations instead of 128… in our case, it's a game changer for performances 😄 👍

### CODE REVIEW

**Review by Siwayll — APPROVED** [review]
:astonished:

File: Entry/AbstractIP.php [hunk 1]
```diff
@@ -173,12 +173,21 @@ protected function IPLongCompare($long1, $long2, $operator = '=')
...
-        $result = 0;
-        for ($bit = 0; $bit < static::NB_BITS; $bit++) {
-            $div = bcpow(2, $bit);
-            $and = bcmod(bcdiv($long1, $div, 0), 2) && bcmod(bcdiv($long2, $div, 0), 2);
-            $result = bcadd($result, bcmul($and, $div));
+        $result = '0';
+        $i = 0;
+
+        // As soon as a number is 0, the result of a bitwise-& cannot change.
+        while ($long1 && $long2) {
```
Comment (Slashgear) [inline Entry/AbstractIP.php]: Maybe we could pass this condition as a param of a factory function in order to reduce this duplicate code ?

Comment (b-viguier) [reply Entry/AbstractIP.php]: ⚠️ Lines 186 and 214 are different! `&` vs `|`
So I'm not convinced by the gain of a factory 😕

**Comment by b-viguier** [comment]
⚠️ 🚫 
`PHP_INT_MAX` is **not** a power of 2, so the algorithm cannot work. I will push ASAP a new version working with the highest power of 2 below `PHP_INT_MAX`, and maybe some tests to check these edge cases

**Comment by b-viguier** [comment]
* Operations made with a power of 2 (to work correctly with big ranges)
* Test added with big ranges

**Review by e-zannelli — APPROVED** [review]

**Review by valentin-claras — APPROVED** [review]

**Comment by b-viguier** [comment]
Go for `v1.0.3`
